appsec/src/extension/ip_extraction.c

```diff
@@ -239,27 +269,43 @@ static bool _parse_ip_address(
 static bool _parse_ip_address_maybe_port_pair(
     const char *nonnull addr, size_t addr_len, ipaddr *nonnull out);
 
-static bool _parse_x_forwarded_for(
+static struct extract_res _parse_multiple_maybe_port(
     zend_string *nonnull zvalue, ipaddr *nonnull out)
 {
     const char *value = ZSTR_VAL(zvalue);
     const char *end = value + ZSTR_LEN(zvalue);
-    bool succ;
+    ipaddr first_private = {0};
     do {
         for (; value < end && *value == ' '; value++) {}
         const char *comma = memchr(value, ',', end - value);
         const char *end_cur = comma ? comma : end;
-        succ = _parse_ip_address_maybe_port_pair(value, end_cur - value, out);
+        ipaddr cur;
+        bool succ =
+            _parse_ip_address_maybe_port_pair(value, end_cur - value, &cur);
         if (succ) {
-            succ = !_is_private(out);
+            if (!_is_private(&cur)) {
+                memcpy(out, &cur, sizeof *out);
+                return EXTRACT_SUCCESS_PUBLIC;
+            }
+            // else is private
```

Comment:
I think the previous conditional is readable enough that this comment is not adding anything